api/src/main/java/org/bf2/operator/resources/v1alpha1/SecretKeySelector.java

```diff
@@ -0,0 +1,88 @@
+
+package org.bf2.operator.resources.v1alpha1;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@JsonDeserialize(using = com.fasterxml.jackson.databind.JsonDeserializer.None.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+        "apiVersion",
+        "kind",
+        "metadata",
+        "key",
+        "name",
+        "optional"
+})
+@ToString
+@EqualsAndHashCode
+@Buildable(editableEnabled = false, validationEnabled = false, lazyCollectionInitEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
+public class SecretKeySelector implements KubernetesResource
+{
+
+    @JsonProperty("key")
+    private String key;
+    @JsonProperty("name")
+    private String name;
+    @JsonProperty("optional")
+    private Boolean optional;
+
```

Comment:
what is the issue with `addtionalProperties` here that we had to clone this class?

Reply:
It's annotated with `@JsonIgnore` which has a `value` attribute that's a `boolean`. The version of fabric8 we're using reads `value` generically and casts it to a string without checking the type first.